src/components/TagButton.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useState } from "react";
+import { Tag } from "../types/Tag";
+import { ListItem } from "../types/ListItem";
+import TagList from "./TagList";
+import { errorTags } from "../functions/errorMessages";
+import {
+  Button,
+  TextField,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+} from "@material-ui/core";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+
+export type TagButtonProps = {
+  item: ListItem;
+  knownTags: Tag[];
+  tagAdded: (item: ListItem, tag: Tag) => void;
+  tagDeleted: (item: ListItem, tag: Tag) => void;
+  updateKnownTags: (title: string) => Tag | undefined;
+};
+export const TagButton: (props: TagButtonProps) => JSX.Element = ({
+  item,
+  knownTags,
+  tagAdded,
+  tagDeleted,
+  updateKnownTags,
+}: TagButtonProps) => {
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const [errorTag, setErrorTag] = useState("");
+  const isTagNew = function (tagTitle: string, tags: Tag[]): Tag | undefined {
+    return tags.find((t) => t.title === tagTitle);
+  };
+  const onChangeAutocomplete = (
+    value: Tag[],
+    reason: string,
+    item: ListItem
+  ) => {
+    const targetTag = value[value.length - 1];
+    if (reason === "select-option") {
+      const tagAlreadyAssigned = isTagNew(targetTag.title, item.tags);
+      if (tagAlreadyAssigned) {
+        () => setErrorTag(errorTags.submit);
```

Comment:
```suggestion
        setErrorTag(errorTags.submit);
```
not sure but I think we just want to call the function here.